### PR TITLE
Feature/add smoke tests api db

### LINKED ISSUE #3: Add smoke tests for services
Description:
We need to implement smoke tests to verify that our core services are up and running after deployment. This will help ensure the reliability of our deployments and quickly identify any issues.

Tasks:
1. Create a smoke test for the API server
   - Implement a simple health check endpoint in the API
   - Write a test that calls this endpoint and verifies the response
   - Connect to a taskfile
2. Create a smoke test for the Postgres database
   - Implement a basic database connection check
   - Write a test that attempts to connect to the database and perform a simple query
   - Connect to a taskfile

There should be a command that runs both integration smoke tests 

Things to consider in the future: 
Integrate these tests into our deployment pipeline
   - Add a step in the deployment process to run these smoke tests
   - Configure the pipeline to fail if the smoke tests don't pass

### PR DESCRIPTION
**Pull request by PrateekKumar1709**
## Problem Statement
We need to implement smoke tests to verify that our core API and database services are up and running after deployment. This will help ensure the reliability of our deployments and quickly identify any issues.

## Related Issue
Fixes #3  - Add smoke tests for services

### API Smoke Test
1. Create a new health check endpoint in `modules/odr_api/odr_api/api/endpoints/health.py`
2. Update `modules/odr_api/odr_api/api/main.py` to include the health check router
3. Implement a test in `modules/odr_api/tests/integration/models/test_health.py`
4. Update `modules/odr_api/Taskfile.yml` to include a smoke test task

### Database Smoke Test
1. Implement a database connection test in `modules/odr_database/tests/test_db_connection.py`
2. Update `modules/odr_database/Taskfile.yml` to include a smoke test task

### Combined Smoke Tests
1. Update the root `Taskfile.yml` to include a task that runs both API and database smoke tests

### CODE REVIEW

File: modules/odr_api/Taskfile.yml [hunk 1]
```diff
@@ -15,3 +15,8 @@ tasks:
     cmds:
       - cmd: python -m debugpy --listen 0.0.0.0:5678 --wait-for-client ./modules/odr_api/odr_api/main.py
         platforms: [windows, linux, darwin]
+
+  test_api:
```
Comment (fearnworks) [inline modules/odr_api/Taskfile.yml:19]: This should be included in the api-test taskfile https://github.com/Open-Model-Initiative/OMI-Data-Pipeline/blob/main/modules/odr_api/tests/Taskfile.yml . 

To keep naming in alignment across tasks use - and not _ in task names, (change to test-api). I would also just rename this one to health-check and add it as the first command ran in the test-all command in the odr_api/test/Taskfile.yml

Comment (fearnworks) [reply modules/odr_api/Taskfile.yml:19]: Seems like something is a bit off when running this command when in the root directory: 

```
PS C:\OMI-Data-Pipeline> task api:test_api
task: [api:test_api] pytest -s modules/odr_api/tests/integration/models/test_health.py
========================================= test session starts ==========================================
platform win32 -- Python 3.12.0, pytest-8.3.2, pluggy-1.5.0
rootdir: C:\OMI-Data-Pipeline\modules\odr_api
configfile: pyproject.toml
plugins: anyio-4.4.0
collected 0 items / 1 error

================================================ ERRORS ================================================ 
_______________________ ERROR collecting tests/integration/models/test_health.py _______________________ 
ImportError while importing test module 'C:\OMI-Data-Pipeline\modules\odr_api\tests\integration\models\test_health.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
C:\AppData\Local\Programs\Python\Python312\Lib\importlib\__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
modules\odr_api\tests\integration\models\__init__.py:1: in <module>
    from models.test_users import TestUserLifecycle
E   ModuleNotFoundError: No module named 'models'
======================================= short test summary info ======================================== 
ERROR modules\odr_api\tests\integration\models\test_health.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! 
=========================================== 1 error in 0.08s =========================================== 
task: Failed to run task "api:test_api": exit status 2
PS C:\OMI-Data-Pipeline>
```

File: modules/odr_api/odr_api/main.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def test_communication():
 app.include_router(annotation_router, prefix=settings.API_V1_STR)
 app.include_router(auth_router, prefix=settings.API_V1_STR)
 app.include_router(embedding_router, prefix=settings.API_V1_STR)
+app.include_router(health.router)
```
Comment (fearnworks) [inline modules/odr_api/odr_api/main.py:34]: Try and match to the existing routes. Should have a prefix and same naming style

File: modules/odr_database/tests/test_db_connection.py
Comment (fearnworks) [inline modules/odr_database/tests/test_db_connection.py:1]: This test seems fine!

**Review by fearnworks — CHANGES_REQUESTED** [review]
Some issues that need to be addressed for the api test before merging. But looks mostly good.

**Comment by PrateekKumar1709** [comment]
Thank you for the review @fearnworks. I've addressed all the points raised and made the necessary changes. Please let me know if anything else needs attention

**Comment by fearnworks** [comment]
Had to make the changes from the last two commits to get running :  https://github.com/Open-Model-Initiative/OMI-Data-Pipeline/tree/feature/add-smoke-tests-api-db

**Comment by fearnworks** [comment]
If you can get those integrated into your PR I think we are good to merge this in.

**Review by fearnworks — APPROVED** [review]
Reviewed, tests performed and merged into PR by author. Merging

**Pull request merged**